.github/workflows/tr.unit.tests.yaml

```diff
@@ -0,0 +1,50 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   https://apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Traffic Router Unit Tests
+
+on:
+  push:
+    paths:
+      - traffic_router/**
+  create:
+  pull_request:
+    paths:
+      - traffic_router/**
+    types: [ opened, reopened, edited, synchronize ]
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@master
+      - name: Cache local Maven repository
+        uses: actions/cache@v2
+        with:
+          path: ${{ github.workspace }}/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+      - name: Run unit tests
+        uses: ./.github/actions/tr-tests
+      - name: Upload Surefire Reports
+        uses: actions/upload-artifact@v2
+        if: ${{ always() }}
+        with:
+          name: surefire-reports
+          path: ${{ github.workspace }}/traffic_router/core/target/surefire-reports/*
```

Comment:
Does it output GHA Annotations or are surefire reports the only rich view of failures we're gonna get?

Reply:
Okay, outputting GHA annotations in 95da3063bd.